### PR TITLE
return early instead of nesting in metrics.py

### PR DESCRIPTION
**Pull request by johnwlambert**

### CODE REVIEW

File: gtsfm/utils/metrics.py [hunk 1]
```diff
@@ -61,14 +61,19 @@ def compute_correspondence_metrics(
...
         return None, None
+    
+    if gt_wTi1 is None or gt_wTi2 is None:
+        return None, None
 
     # Compute ground truth correspondences.
     matched_keypoints_i1 = keypoints_i1.extract_indices(corr_idxs_i1i2[:, 0])
     matched_keypoints_i2 = keypoints_i2.extract_indices(corr_idxs_i1i2[:, 1])
-    if None not in [gt_scene_mesh, gt_wTi1, gt_wTi2]:
+    # check to see if a GT mesh is provided.
```
Comment (travisdriver) [inline gtsfm/utils/metrics.py:75]: nit: capitalize first letter of sentence

File: gtsfm/utils/metrics.py [hunk 2]
```diff
@@ -79,19 +84,18 @@ def compute_correspondence_metrics(
...
-            intrinsics_i2,
-            gt_i2Ti1,
-            dist_threshold,
-        )
-    else:
-        return None, None
-
+        return is_inlier, reproj_error
+        
+    # no mesh is provided, so we use squared Sampson error.
```
Comment (travisdriver) [inline gtsfm/utils/metrics.py:89]: nit: word comments as commands, e.g., "If no mesh is provided, use squared Sampson distance."
nit: capitalize first letter of sentence

File: gtsfm/utils/metrics.py (hunk 1 above)
Comment (johnwlambert) [reply gtsfm/utils/metrics.py]: thanks, updated.

File: gtsfm/utils/metrics.py (hunk 2 above)
Comment (johnwlambert) [reply gtsfm/utils/metrics.py]: thanks, updated.

**Review by travisdriver — COMMENTED** [review]
LGTM

**Review by travisdriver — APPROVED** [review]

**Pull request merged**